### ISSUE TITLE
Should we deprecate `LevMarLSQFitter`?

### ISSUE THREAD

**Issue by Cadair** [astropy/astropy#16607]
### What is the problem this feature will solve?

We have two implementations of the Levenberg-Marquardt fitting algortihm [LevMarLSQFitter](https://docs.astropy.org/en/stable/api/astropy.modeling.fitting.LevMarLSQFitter.html#astropy.modeling.fitting.LevMarLSQFitter) and [LMLSQFitter](https://docs.astropy.org/en/stable/api/astropy.modeling.fitting.LMLSQFitter.html#astropy.modeling.fitting.LMLSQFitter). The latter uses a newer function in scipy [`least_squares`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html) and the former uses the older [`leastsq`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.leastsq.html).

It's exceptionally not-obvious from the documentation why you might choose one over the other, in fact, I had to spend a good couple of minutes figuring out which was which and I *knew* one was a newer implementation in scipy.

I am sure there are still a lot of uses of the old one in the wild, so deprecating it might be tricky, but perhaps we could make an extra long deprecation period or something. Or at least explicitly recommend the newer one in the docs?

**Comment by Cadair** [comment]
Further spelunking reveals that there is a subtle (and problematic) difference between the old `LevMarLSQFitter` and the new `LMLSQFitter` in that the old one has hacky support for bounds on the fit, but that can cause issues as reported in https://github.com/astropy/astropy/issues/11946 so we should make that clearer in the docs as well.

**Comment by keflavich** [comment]
`LevMarLSQFitter` is the cornerstone of a lot of codes I've used, written, and seen around.  I'm :-1: on deprecating it because that will create a huge maintenance burden for me specifically, and I think it will make a lot of unmaintained codes unusable in a couple years.  I'm hugely in favor of documenting the issues you noted, though, and clarifying when/why I should switch!

**Comment by pllim** [comment]
I don't think specutils would like it either but maybe that is one of the "cornerstone" @keflavich mentioned above.

https://github.com/astropy/specutils/blob/d16d91a077417a7aacdee06d2b259fa887908b14/specutils/fitting/fitmodels.py#L261

cc @rosteen

**Comment by pllim** [comment]
Should we forget about deprecation and change the scope to doc update?

**Comment by keflavich** [comment]
IMO, "it's been here for over a decade" is a reason to keep it around.  It's closely integrated into a lot of publications and is part of the literature.  But I acknowledge the counterargument that every line of code implies some amount of maintenance cost.

**Comment by keflavich** [comment]
Yeah I agree with that.  I think docstrings of both functions needs to point to the (new?) document that describes how they're different.  But you're right that that's going to have limited impact.

**Comment by rosteen** [comment]
> I don't think specutils would like it either but maybe that is one of the "cornerstone" @keflavich mentioned above.
> 
> https://github.com/astropy/specutils/blob/d16d91a077417a7aacdee06d2b259fa887908b14/specutils/fitting/fitmodels.py#L261
> 
> cc @rosteen

I don't feel strongly either way, refactoring in specutils would be minimal effort, and in fact I was aware of the newer fitter and at least at some point had thought of/started to replace the old one. I don't recall why I didn't, honestly, maybe there was some unexpected wrinkle.

**Comment by WilliamJamieson** [comment]
My intention was to eventually deprecate it. However, if it becomes a maintenance issue I can see having it alias the `LMLSQFitter`. They should agree up to some minor numerical differences except when having bounded parameter values. The question is how to go about informing users of the tiny change.
